ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -92,15 +104,53 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 		containerDefinitions = append(containerDefinitions, containerDef)
 	}
 
+	var ecsFields *ECSFields
+	if ecsFieldsFileName != "" {
+		ecsFields, _ = readECSFields(ecsFieldsFileName)
+	}
+
+	var networkMode string
+	if ecsFields != nil {
+		networkMode = ecsFields.TaskDefinition.NetworkMode
+		// The task-role-arn flag should take precedence over a taskRoleArn value specified in ECS fields file
+		if taskRoleArn == "" {
+			taskRoleArn = ecsFields.TaskDefinition.TaskRoleArn
+		}
+	}
+
 	taskDefinition := &ecs.TaskDefinition{
 		Family:               aws.String(taskDefinitionName),
 		ContainerDefinitions: containerDefinitions,
 		Volumes:              convertToECSVolumes(volumes),
 		TaskRoleArn:          aws.String(taskRoleArn),
+		NetworkMode:          aws.String(networkMode),
 	}
+
 	return taskDefinition, nil
 }
 
+func readECSFields(filename string) (*ECSFields, error) {
+	ecsFieldsData, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.WithFields(log.Fields{
```

Comment:
I'd suggest using errors.Wrapf to add the filename to the error. Then simply return it instead of logging it. The top level command function should log it.